src/test/java/uk/gov/digital/ho/proving/income/hmrc/HmrcClientTest.java

```diff
@@ -209,12 +229,80 @@ public void shouldLogEventType() {
             LocalDate.of(2017, Month.JANUARY, 1),
             LocalDate.of(2017, Month.JULY, 1)
         );
-        verify(mockAppender).doAppend(argThat(argument -> {
-            LoggingEvent loggingEvent = (LoggingEvent) argument;
 
-            return loggingEvent.getFormattedMessage().equals("About to call Income Service at http://income-service/income") &&
-                ((ObjectAppendingMarker) loggingEvent.getArgumentArray()[1]).getFieldName().equals("event_id");
-        }));
+        verifyLogMessage("Received 0 incomes and 0 employments", HMRC_RESPONSE_SUCCESS, INFO, 2);
+    }
+
+    @Test
+    public void shouldLogWhenHmrcNotFoundResponseReceived() {
+        when(mockRestTemplate.exchange(anyString(), any(HttpMethod.class), any(HttpEntity.class), ArgumentMatchers.<Class<IncomeRecord>>any()))
+            .thenThrow(new HttpClientErrorException(NOT_FOUND));
+
+        try {
+            service.getIncomeRecord(
+                new Identity(
+                    "John",
+                    "Smith",
+                    LocalDate.of(1965, Month.JULY, 19), "NE121212A"),
+                LocalDate.of(2017, Month.JANUARY, 1),
+                LocalDate.of(2017, Month.JULY, 1)
+            );
+        }
+        catch(EarningsServiceNoUniqueMatchException e){
+            //not used for the purpose of this test
+        }
+
+       verifyLogMessage("HMRC Service found no match", HMRC_NOT_FOUND_RESPONSE, ERROR, 2);
+    }
+
+    @Test
+    public void shouldLogWhenHmrcServiceFails() {
+        when(mockRestTemplate.exchange(anyString(), any(HttpMethod.class), any(HttpEntity.class), ArgumentMatchers.<Class<IncomeRecord>>any()))
+            .thenThrow(new HttpServerErrorException(INTERNAL_SERVER_ERROR));
+
+        try {
+            service.getIncomeRecord(
+                new Identity(
+                    "John",
+                    "Smith",
+                    LocalDate.of(1965, Month.JULY, 19), "NE121212A"),
+                LocalDate.of(2017, Month.JANUARY, 1),
+                LocalDate.of(2017, Month.JULY, 1)
+            );
+        }
+        catch(HttpServerErrorException e){
+            //not used for the purpose of this test
+        }
+
+        verifyLogMessage("HMRC Service failed", HMRC_ERROR_REPSONSE, ERROR, 2);
+    }
+
+    @Test
+    public void shouldLogOnIncomeRecordFailureRecovery() {
+        HttpServerErrorException exception = new HttpServerErrorException(BAD_GATEWAY);
+
+        try {
+            service.getIncomeRecordFailureRecovery(exception);
+        }
+        catch(HttpServerErrorException e) {
+            //not used for the purpose of this test
+        }
+
+        verifyLogMessage("Failed to retrieve HMRC data after retries - 502 BAD_GATEWAY", HMRC_ERROR_REPSONSE, ERROR, 1);
+    }
+
+    public void verifyLogMessage(String message, LogEvent event, Level logLevel, int invocations) {
```

Comment:
I don't think the invocations has added anything useful -  I'd refactor to remove it. You should be able to do something like 
verify(mockAppender, atLeastOnce()).doAppend(captor.capture())
and then assert that one of the arguments looks like the expected.

Or write it how we do it in other tests:
verify(mockAppender, atLeastOnce()).doAppend(argThat ->{
...
})

Reply:
Refactored to remove the `invocations` parameter and included the `atLeastOnce()` method call.